aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -570,6 +573,215 @@ struct cpu_scatter_gather_base_kernel {
   }
 };
 
+template <typename scalar_t, SCATTER_GATHER_OP reduce>
+inline void init(scalar_t* ptr, int64_t size, bool include_self) {
+  if (!include_self) {
+    using acc_t = vec::vec_scalar_t<scalar_t>;
```

Comment:
Hmm, don't we need something like `acc_scalar_t` (I.e. I assume for bfloat16 it should be `float`, shouldn't it?)
If not, please add comment why it does not follow the same pattern

Reply:
using `at::acc_type<scalar_t, false>;` has a major problem that it points to `double` for `scalar_t=float`, which is not expected for vectorization purpose. This is why i wrote `vec_scalar_t<>` at the beginning with `vec_scalar_t<BFloat16> = float`.

`vec_scalar_t<>` is now kind of a duplicate for `at::opmath_type<scalar_t>;` And we will replace `vec_scalar_t` with `opmath_type` eventually. For example, this one: https://github.com/pytorch/pytorch/pull/91086.

The only problem is that `vec_scalar_t` did not handle `Half`, so `vec::map<>` from `functional.py` will run into compilation error if I directly replace `vec_scalar_t` with `opmath_type`.

Anyway, I will fix this and eventually we will remove `vec_scalar_t` and use `opmath_type` instead.